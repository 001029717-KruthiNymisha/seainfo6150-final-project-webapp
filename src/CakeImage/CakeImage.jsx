import React from "react";
import styles from "./CakeImage.module.css"

const CakeImage = ({url, title}) => {
  return (
    <div className = {styles.cakeImage}>
      <img src={url} alt={title} />
    </div>
  );
};

export default CakeImage;