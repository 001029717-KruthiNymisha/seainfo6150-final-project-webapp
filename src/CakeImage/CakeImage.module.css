.cakeImage {
  width: 400px;
  height: 400px;
  overflow: hidden;
  position: relative;
  margin: 10px;
  margin-top: 20px;
  margin-bottom: 40px;
  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);
}

.cakeImage img {
object-fit: cover;
  display: block;
  max-width: 100%;
  max-height:100%;
  margin-left:0px;
  margin-right:-10px;
  margin-top:0px;
  margin-bottom:9px;
  position: absolute;
  left: 0;
  top:50%;
  transform: translateY(-50%);
}
@media (min-width: 400px) {
.cakeImage{
width:200px;
height:200px;
}
}
@media (min-width: 800px) {
.cakeImage{
width:300px;
height:300px;
}
}
@media (min-width: 1000px) {
.cakeImage{
width:400px;
height:400px;
}
}