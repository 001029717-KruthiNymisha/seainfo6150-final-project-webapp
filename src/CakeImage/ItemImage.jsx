import React from "react";
import css from "./ItemImage.module.css"

const ItemImage = ({url, title}) => {
  return (
    <div className = {css.itemImage}>
      <img src={url} alt={title} />
    </div>
  );
};

export default ItemImage;