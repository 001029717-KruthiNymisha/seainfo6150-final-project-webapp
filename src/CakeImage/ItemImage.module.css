.itemImage {
  width: 500px;
  height: 490px;
  margin: 10px;
}

.itemImage img {
  padding: 10px;
  max-width: 100%;
  max-height: 100%;
  margin: auto;
  margin-top:-15px;
}
@media(min-width:400px){
.itemImage{
   width: 400px;
   height: 380px;
}
.itemImage img{
 max-width: 100%;
  max-height: 100%;
  margin-left:50px;
}
}
@media(min-width:800px){
.itemImage{
   width: 450px;
   height: 420px;
}
.itemImage img{
 max-width: 100%;
  max-height: 100%;
  margin-left:70px;
}
}
@media(min-width:1000px){
.itemImage{
   width: 500px;
   height: 490px;
}
.itemImage img{
 max-width: 100%;
  max-height: 100%;
  margin-left:100px;
}
}