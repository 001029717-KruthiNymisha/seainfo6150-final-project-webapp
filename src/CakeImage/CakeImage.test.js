import React from "react";
import { render } from "@testing-library/react";
import CakeImage from "./CakeImage.jsx";

it("renders the CakeImage component correctly", () => {
  const url="https://assets.wsimgs.com/wsimgs/rk/images/dp/wcm/202038/0588/img36o.jpg";

  const { container } = render(<CakeImage url={url} />);
  expect(container).toMatchSnapshot();
});