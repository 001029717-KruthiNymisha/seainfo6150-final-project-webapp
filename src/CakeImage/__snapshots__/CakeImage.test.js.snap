// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the CakeImage component correctly 1`] = `
<div>
  <div
    class="cakeImage"
  >
    <img
      src="https://assets.wsimgs.com/wsimgs/rk/images/dp/wcm/202038/0588/img36o.jpg"
    />
  </div>
</div>
`;
